zarr 3.2.0 breaks saving empty Dask arrays

CI for Python >=3.12 broke after zarr 3.2.0 was released:
> FAILED dask/array/tests/test_array_core.py::test_to_zarr_accepts_empty_array_without_exception_raised - ValueError: Dimension 0: integer chunk edge length must be >= 1, got 0

e.g. https://github.com/dask/dask/actions/runs/25366857340/job/74379837613?pr=12341

Python 3.10 and 3.11 are green because they are stuck at zarr 3.1.6.

--- a/dask/array/core.py
+++ b/dask/array/core.py
@@ -4095,7 +4095,8 @@ def to_zarr(
     zarr_store = _setup_zarr_store(url, storage_options, mode=mode)
 
     zarr_array_kwargs.setdefault("shape", arr.shape)
-    zarr_array_kwargs.setdefault("chunks", tuple(c[0] for c in arr.chunks))
+    # Empty Dask arrays have chunk size 0, which zarr does not accept
+    zarr_array_kwargs.setdefault("chunks", tuple(max(c[0], 1) for c in arr.chunks))
     zarr_array_kwargs.setdefault("dtype", arr.dtype)
 
     array_name = component or zarr_array_kwargs.pop("name", None)

--- a/dask/array/tests/test_array_core.py
+++ b/dask/array/tests/test_array_core.py
@@ -5014,11 +5014,20 @@ def test_zarr_roundtrip_with_path_like(tmp_path, zarr_format):
     assert a2.chunks == a.chunks
 
 
-def test_to_zarr_accepts_empty_array_without_exception_raised():
+@pytest.mark.parametrize(
+    "a",
+    [
+        da.arange(0),  # shape=(0,); chunks=((0,),)
+        da.array([[1]])[:0],  # shape=(0, 1); chunks=((0,), (1,))
+        da.array([[1]])[:, :0],  # shape=(1, 0); chunks=((1,), (0,))
+    ],
+)
+def test_zarr_empty_array(tmp_path, a):
     pytest.importorskip("zarr")
-    with tmpdir() as d:
-        a = da.from_array(np.arange(0))
-        a.to_zarr(d)
+    fname = tmp_path / "x.zarr"
+    a.to_zarr(fname)
+    b = da.from_zarr(fname)
+    assert_eq(a, b)
 
 
 @pytest.mark.parametrize("compute", [False, True])

